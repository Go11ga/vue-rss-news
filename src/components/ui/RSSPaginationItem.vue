<template>
    <router-link 
        class="rss-pagination__link"
        :class="{ 
            'rss-pagination__link--big' : size === 'big',
            'rss-pagination__link--active' : active 
        }"
        :to="path"
    >
        {{ label }}
    </router-link>
</template>

<script>
export default {
    name: 'RSSPaginationItem',

    props: {
        label: {
            type: String,
            required: true
        },
        value: {
            type: Number,
            required: true
        },
        active: {
            type: Boolean,
            default: false
        },
        size: {
            type: String
        }
    },

    computed: {
        path() {
            const { name } = this.$route

            return `/${name}/${this.value}`
        }
    }
}
</script>
