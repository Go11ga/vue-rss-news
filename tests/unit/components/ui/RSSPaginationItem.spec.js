import { shallowMount, createLocalVue } from '@vue/test-utils'
import RSSPaginationItem from '@/components/ui/RSSPaginationItem'
import VueRouter from 'vue-router'

const localVue = createLocalVue()
localVue.use(VueRouter)

describe('RSSPaginationItem.vue', () => {
    let wrapper
    let router

    beforeEach(() => {
        router = new VueRouter()

        wrapper = shallowMount(RSSPaginationItem, {
            localVue,
            router,
            propsData: {
                label: '1',
                value: 1,
                active: true,
                size: 'big'
            }
        })
    })

    afterEach(() => {
        wrapper.destroy()
    })

    it('проверка имени компонента', () => {
        expect(wrapper.vm.$options.name).toMatch('RSSPaginationItem')
    })

    it('snapshot', () => {
        expect(wrapper.element).toMatchSnapshot()
    })

    it('props', async () => {
        expect(wrapper.classes().includes('rss-pagination__link')).toBe(true)
        expect(wrapper.classes().includes('rss-pagination__link--big')).toBe(true)
        expect(wrapper.classes().includes('rss-pagination__link--active')).toBe(true)

        wrapper.setProps({
            label: '1',
            value: 1,
            active: false,
            size: undefined
        })

        await localVue.nextTick()

        expect(wrapper.classes().includes('rss-pagination__link')).toBe(true)
        expect(wrapper.classes().includes('rss-pagination__link--big')).toBe(false)
        expect(wrapper.classes().includes('rss-pagination__link--active')).toBe(false)
    })

    it('computed path', () => {
        let localThis = {
            $route: {
                name: 'mos' 
            },
            value: 1
        }

        expect(RSSPaginationItem.computed.path.call(localThis)).toMatch('/mos/1')
    })
})
