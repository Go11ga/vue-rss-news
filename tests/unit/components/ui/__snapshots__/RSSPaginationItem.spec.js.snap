// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RSSPaginationItem.vue snapshot 1`] = `
<router-link-stub
  ariacurrentvalue="page"
  class="rss-pagination__link rss-pagination__link--big rss-pagination__link--active"
  event="click"
  tag="a"
  to="/null/1"
>
  
        1
    
</router-link-stub>
`;
